### PR TITLE
AArch64: Implement ARM64MonitorEnter/ExitSnippet

### PR DESCRIPTION
**Pull request by knn-k**
This commit implements ARM64MonitorEnterSnippet and
ARM64MonitorExitSnippet.

Signed-off-by: knn-k <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
This uses `ARM64HelperCallSnippet` implemented in https://github.com/eclipse/omr/pull/4015, but I think this PR can be reviewed independently.

**Comment by knn-k** [comment]
Updated to use `directCallRequiresTrampoline()` instead of `constantIsSignedImm28()` for checking branch range.

**Comment by knn-k** [comment]
Implementing the monitorEnter/Exit evaluators that use these snippets.  Added "WIP:" to the title.

**Comment by knn-k** [comment]
Ready for review.

**Comment by knn-k** [comment]
Rebased and resolved the merge conflict.

**Comment by knn-k** [comment]
Updated the code by adding changes for compressed refs, and rebased.

File: runtime/compiler/aarch64/codegen/J9ARM64Snippet.cpp [hunk 1]
```diff
@@ -23,88 +23,218 @@
...
+   *(int32_t *)buffer |= ((4 << 5) | TR::CC_NE); // 4 instructions forward, bne
+   buffer += ARM64_INSTRUCTION_LENGTH;
+
+   *(int32_t *)buffer = TR::InstOpCode::getOpCodeBinaryEncoding(TR::InstOpCode::addimmx);
+   dataReg->setRegisterFieldRD((uint32_t *)buffer);
+   dataReg->setRegisterFieldRN((uint32_t *)buffer);
+   *(int32_t *)buffer |= ((LOCK_INC_DEC_VALUE & 0xFFF) << 10); // imm12
+   buffer += ARM64_INSTRUCTION_LENGTH;
+
+   op = fej9->generateCompressedLockWord() ? TR::InstOpCode::strimmw : TR::InstOpCode::strimmx;
```
Comment (Akira1Saitoh) [inline runtime/compiler/aarch64/codegen/J9ARM64Snippet.cpp]: compilation fails because `fej9` is not declared in this function.

Comment (knn-k) [reply runtime/compiler/aarch64/codegen/J9ARM64Snippet.cpp]: Thank you for pointing it out.  I had put a wrong version before fix into the commit.

**Comment by 0xdaryl** [comment]
Jenkins compile alinux64xl jdk11

**Pull request merged**